### DISCUSSION TITLE
Binding html content from data objects dynamically

### DISCUSSION THREAD

**Discussion by eoehen** [richard-scryber/scryber.core#96] [Q&A]
I'm not sure, but i didn't found a solution in the documentation.
Is there a solution to bind html from data object?

The normal model content binding is not working:

`<div>{@:Model.HtmlContent}</div>`

The content of `Model.HtmlContent` could be:

```
<p><span style='color: rgb(0, 0, 0);'>Lorem ipsum dolor sit amet, consectetur</span></p>
<p><br/></p>
<p><span style='color: rgb(0, 0, 0);'><span class='ql-cursor'></span>Vivamus sed tincidunt nisi. Vivamus elementum,</span></p>

```

Thank you for hint's or feedback.

**Comment by richard-scryber** [comment]
Hey eoehan

That is a very good question, and actually something that has slipped through the net in conversion to using html. I tried a few things, but the best seems to be falling back to the old components.

I have an example below using the component qualified namespace declaration at the top and then the  qualified PlaceHolder within the content. This should work.

```
<html lang='en' xmlns='http://www.w3.org/1999/xhtml'
      xmlns:c="http://www.scryber.co.uk/schemas/core/release/v1/Scryber.Components.xsd" >
<head>
    <title>Bound placeholder content</title>
</head>
<body title="Outer">
    <p style="margin-top:0px; padding:10pt">Below is the dynamic content</p>
    <c:PlaceHolder contents="{@:model.HtmlContent}" />
</body>
</html>
```

If it doesn't work for you then you can follow the example of adding content dynamically in code from the example [here](https://scrybercore.readthedocs.io/en/latest/6_binding/1_binding_model.html#injecting-content).

(Next job is to definitely update the documentation on all of it).

I would love to hear how you think it would be best to add to the xhtml?
It would be simplest as an attribute, and then maybe something like below.

```
<div data-html='{@:model.HtmlContent}' data-html-placement='Append|Prepend|Replace' ></div>
```

Would love your feedback on this.

Richard

**Comment by eoehen** [comment]
Dear Richard

Thank you for your answer.
Unfortunately both suggestions do not work.

This variant is empty in rendered pdf:

`<c:PlaceHolder contents="{@:model.HtmlContent}" />`

[This](https://scrybercore.readthedocs.io/en/latest/6_binding/1_binding_model.html#injecting-content) suggestion don't work for me, because I am within an enumeration of models and each model has html content.

I like your suggestion:

`<div data-html='{@:model.HtmlContent}' data-html-placement='Append|Prepend|Replace' ></div>`

Thank you for your feedback.
Elias

**Reply by richard-scryber** [reply]
OK - I did notice that my 'm' was lowercase to your Model, but if its still not working, can you append the logs and see if there is anything output.

Add the following to the very top of your template...


`<?scryber append-log='true' log-level='Verbose' parser-log='true' ?>`

You should get a few pages at the bottom that show what is happening. If the Placeholder is not being referenced, then this will be shown. If the data is causing an error it will be shown after the data binding starts.

**Reply by richard-scryber** [reply]
Did you manage to get this working Elias?

**Reply by eoehen** [reply]
No unfortunately it did not work. :-(

**Reply by richard-scryber** [reply]
Ok. I’m just working on hyphens, but should be able to sort the binding after that. Monday and Tuesday next week.

**Reply by eoehen** [reply]
Thank you.

**Comment by richard-scryber** [comment]
@eoehen  - I have pushed the Binding-raw-html branch which should support the use of the following attributes on any visual component (Page, Div, Span, etc).

```
data-content='{{model.content}}' //The actual inner content to bind
data-content-type='text/html' //The mime-type for the inner content, default is application/xhtml+xml for xhtml
data-content-action='Append' //What to do with the content. Append (default), Prepend, or Replace
```

For example

`<div data-content='{{model.HtmlContent}}' data-content-action='Replace' ></div>`


I would really appreciate your feedback on this, check it works for you, and then I can merge with the main branch and add to the next release. Thanks in advance. :-)
